src/libvirtApi/domain.js

```diff
@@ -888,6 +888,33 @@ export function domainSendNMI({
     return call(connectionName, objPath, 'org.libvirt.Domain', 'InjectNMI', [0], { timeout, type: 'u' });
 }
 
+function shlex_quote(str) {
+    // yay, command line apis...
+    return "'" + str.replaceAll("'", "'\"'\"'") + "'";
```

Comment:
What about https://docs.python.org/3/library/shlex.html#shlex.quote ?

Reply:
Yeah, what about? :-) This is supposed to be a JavaScript implementation of it. Is something missing?

Reply:
Mistaken this for a Python file, don't ask :)